src/vs/workbench/contrib/chat/browser/chatMarkdownDecorationsRenderer.ts

```diff
@@ -89,21 +95,28 @@ export class ChatMarkdownDecorationsRenderer {
 			} else if (part instanceof ChatRequestAgentPart) {
 				result += this.instantiationService.invokeFunction(accessor => agentToMarkdown(part.agent, false, accessor));
 			} else {
-				const uri = part instanceof ChatRequestDynamicVariablePart && part.data instanceof URI ?
-					part.data :
-					undefined;
-				const title = uri ? encodeURIComponent(this.labelService.getUriLabel(uri, { relative: true })) :
-					part instanceof ChatRequestAgentPart ? part.agent.id :
-						'';
-
-				const text = part.text;
-				result += `[${text}](${decorationRefUrl}?${title})`;
+				result += this.genericDecorationToMarkdown(part);
 			}
 		}
 
 		return result;
 	}
 
+	private genericDecorationToMarkdown(part: IParsedChatRequestPart): string {
+		const uri = part instanceof ChatRequestDynamicVariablePart && part.data instanceof URI ?
+			part.data :
+			undefined;
+		const title = uri ? this.labelService.getUriLabel(uri, { relative: true }) :
+			part instanceof ChatRequestSlashCommandPart ? part.slashCommand.detail :
+				part instanceof ChatRequestAgentSubcommandPart ? part.command.description :
+					part instanceof ChatRequestVariablePart ? (this.chatVariablesService.getVariable(part.variableName)?.description ?? '') :
+						'';
```

Comment:
Getting flashbacks of @joyceerhl's PR. How about this fancy syntax I just learned about?:

```ts
let title: string;
switch (part.constructor) {
  case ChatRequestDynamicVariablePart:
    title = part.data instanceof URI
      ? this.labelService.getUriLabel(part.data, { relative: true })
      : '';
    break;
  case ChatRequestSlashCommandPart:
    title = part.slashCommand.detail;
    break;
  case ChatRequestAgentSubcommandPart:
    title = part.command.description;
    break;
  case ChatRequestVariablePart:
    title = this.chatVariablesService.getVariable(part.variableName)?.description ?? ''
    break;

}
```

Reply:
It's `break;`ing my eyes